File: C&C++/Binary2Decimal.c
#include <stdio.h>
#include <string.h>
#include <math.h>

long long B2D(const char *s) {
    long long result = 0;
    int len = strlen(s);
    
    for (int i = 0; i < len; i++)
        result = (result << 1) + (s[i] - '0'); // shift left and add bit

    return result;
}

int main() {
    const char *binary = "1010101001010101010101";
    printf("Binary: %s\nDecimal: %lld\n", binary, B2D(binary));
    return 0;
}

File: JAVA/Binary2Decimal.java
public class Binary2Decimal
{
    public long B2D(String s)
    {
        long d = 0;
        for (long i = s.length()-1; i>=0; i--)
            d += ((long)s.charAt((int)i) - (long)'0')*Math.pow(2,s.length()-1-i);
        return d;
    }
    
    public static void main(String args[])
    {
        Binary2Decimal obj = new Binary2Decimal();
        String str = "1010101001010101010101";
        System.out.println("Binary: "+str+"\nDecimal: "+obj.B2D(str));
    }
}